sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -426,27 +468,9 @@ class SdkSynchronizer private constructor(
     }
 
     override suspend fun refreshExchangeRateUsd() {
-        Twig.info { "Bootstrapping Tor client for fetching exchange rates" }
-        val tor =
-            try {
-                TorClient.new(torDir)
-            } catch (e: Exception) {
-                Twig.error(e) { "Failed to bootstrap Tor client" }
-                throw e
-            }
-
-        val rate =
-            try {
-                tor.getExchangeRateUsd()
-            } catch (e: Exception) {
-                Twig.error(e) { "Failed to fetch exchange rate through Tor" }
-                throw e
-            } finally {
-                tor.dispose()
-            }
-        Twig.info { "Latest USD/ZEC exchange rate is $rate" }
-
-        _exchangeRateUsd.value = Pair(rate, Instant.now())
+        coroutineScope.launch {
+            refreshExchangeRateUsd.emit(Unit)
+        }
```

Comment:
I'm now confused by what this does. Is `refreshExchangeRateUsd()` no longer a manual refresh action, but instead kicks off a stream of automated refreshes (which would explain why errors are no longer raised from here)? If that is the case, then the method name and documentation is incorrect, and should have been altered in #1537 (to something like `startAutomaticRefreshOfExchangeRateUsd`).

Reply:
I see. It still is what the name says: `refreshExchangeRateUsd`. When it fires it signalizes a call to refresh then `exchangeRateUsd`.

How it works:
`refreshExchangeRateUsd` is observed by `exchangeRateUsd` but there is a catch.
`private val refreshExchangeRateUsd = MutableSharedFlow<Unit>(replay = 1)` is a hot flow with a replay cache of 1, but this time it's empty. Because it is empty observing`refreshExchangeRateUsd.flatMapLatest` would not fire.
This is solved by:
`private val refreshExchangeRateUsd = MutableSharedFlow<Unit>(replay = 1).apply { tryEmit(Unit) }`

Now according to documentation:
```
public inline fun <T, R> Flow<T>.flatMapLatest(
    @BuilderInference crossinline transform: suspend (value: T) -> Flow<R>
): Flow<R>

Returns a flow that switches to a new flow produced by transform function every time the original flow emits a value. When the original flow emits a new value, the previous flow produced by transform block is cancelled.
```

So making `refreshExchangeRateUsd` fire again in effect makes the suspending function UsdExchangeRateFetcher.invoke be killed because we are calling it within observing it via`flatMapLatest`.

Also another welcome and intended change is: You can just start collecting and do not have to call `refreshExchangeRateUsd()` manually in order for the tor fetcher to be called. It happens automatically.

Reply:
Mmk, so the effect of this change is:
- On first subscribe to `exchangeRateUsd`, a refresh occurs automatically.
- All subsequent refreshes are individually triggered by `refreshExchangeRateUsd`

That seems fine to me.